sdc/datatypes/common_functions.py

```diff
@@ -641,3 +644,59 @@ def _almost_equal_impl(x, y):
         return abs(x - y) <= numpy.finfo(common_dtype).eps
 
     return _almost_equal_impl
+
+
+def sdc_reindex_series(arr, index, name, by_index):
+    pass
+
+
+@sdc_overload(sdc_reindex_series, jit_options={'parallel': True})
+def sdc_reindex_series_overload(arr, index, name, by_index):
+    """ Reindexes series data by new index following the logic of pandas.core.indexing.check_bool_indexer """
+
+    data_dtype, index_dtype = arr.dtype, index.dtype
+    data_is_str_arr = isinstance(arr.dtype, types.UnicodeType)
+
+    def sdc_reindex_series_impl(arr, index, name, by_index):
+        res_index = by_index.copy()
+        if data_is_str_arr == True:  # noqa
+            res_data_as_list = [''] * len(by_index)
+            res_data_nan_mask = numpy.zeros(len(by_index), dtype=types.bool_)
+        else:
+            res_data = numpy.empty(len(by_index), dtype=data_dtype)
+
+        # build a dict of self.index values to their positions:
+        map_index_to_position = Dict.empty(
+            key_type=index_dtype,
+            value_type=types.int32
+        )
+
+        for i, value in enumerate(index):
+            if value in map_index_to_position:
+                raise ValueError("cannot reindex from a duplicate axis")
+            else:
+                map_index_to_position[value] = i
+
+        index_mismatch = 0
+        for i in numba.prange(len(by_index)):
+            if by_index[i] in map_index_to_position:
+                pos_in_self = map_index_to_position[by_index[i]]
+                res_data[i] = arr[pos_in_self]
+                if data_is_str_arr == True:  # noqa
+                    res_data_nan_mask[i] = isna(arr, i)
```

Comment:
Looks `res_data` isn't defined in case when `data_is_str_arr == True`. So you could add else-block where you could fill `res_data`.

Reply:
@densmirn Thank you! It seems that this case (string series reindexing) is not used in SDC functions implementation yet (getitem uses only bool series), hence it's slipped through functional tests. And that's a case that shows how much we need unit tests for internal functions.